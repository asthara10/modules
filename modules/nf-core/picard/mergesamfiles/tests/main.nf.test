nextflow_process {

    name "Test Process PICARD_MERGESAMFILES"
    script "../main.nf"
    process "PICARD_MERGESAMFILES"

    tag "modules"
    tag "modules_nfcore"
    tag "picard"
    tag "picard/mergesamfiles"

    test("test_picard_mergesamfiles") {


        when {
            process {
                """
                
                input[0] = [ [ id:'test', single_end:false ], // meta map
              [ file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true), ]
            ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

}
