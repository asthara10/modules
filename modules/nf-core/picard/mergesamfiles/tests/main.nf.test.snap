{
    "versions": {
        "content": [
            [
                "versions.yml:md5,85a5ae7f1cbbedb821fb94713e84612c"
            ]
        ],
        "timestamp": "2023-11-24T13:15:51.446142"
    }
}